File: src/main/java/com/talenco/tasksystem/service/StepService.java
package com.talenco.tasksystem.service;

import com.talenco.tasksystem.entity.Step;

import java.util.List;

public interface StepService {

    List<Step> getAll();

    Step getOne(Long id);

    void insert(Step step);

    void update(Step step);

    /**
     * 在删除项目时,删除项目下的步骤,该功能不影响主要供能,可后续完成
     * @param ids 项目id数组
     */
    void delete(Long[] ids);

    List<Step> searchByProjectId(Long projectId);

    List<Step> searchByCheckMan(String username);
}

File: src/main/webapp/js/service/stepService.js
app.service('stepService',function($http){

    // 任务列表
    this.getAll=function(){
        return $http.get('../StepController/getAll');
    }
    // 任务列表
    this.getOne=function(taskId){
        return $http.get('../StepController/getOne?taskId='+taskId);
    }
    // 任务列表
    this.searchByUsername=function(username){
        return $http.get('../StepController/searchByUsername?username='+username);
    }
    // 任务列表
    this.searchByProjectId=function(projectId){
        return $http.get('../StepController/searchByProjectId?projectId='+projectId);
    }
    // 新建任务
    this.insert=function(entity){
        return $http.post('../StepController/insert',entity);
    }
    // 删除用户
    this.dele=function(selectIds){
        return $http.get('../StepController/delete?ids='+selectIds);
    }
    // 更新任务信息
    this.update=function(entity){
        return $http.post('../StepController/update',entity);
    }
})